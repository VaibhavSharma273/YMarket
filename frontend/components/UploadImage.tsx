import React, { useState, useEffect, Component, } from 'react';
import { Image, View, Platform, TouchableOpacity, Text, StyleSheet, Alert } from 'react-native';
import { AntDesign } from '@expo/vector-icons';
import * as ImagePicker from 'expo-image-picker';
import { MaterialIcons } from '@expo/vector-icons'; 
import { transparent } from 'react-native-paper/lib/typescript/styles/colors';

export default function UploadImage({updateImages, defaultValue, number}: any) {
    const [image, setImage] = useState(defaultValue);
    const addImage = async () => {
        let _image = await ImagePicker.launchImageLibraryAsync({
        mediaTypes: ImagePicker.MediaTypeOptions.Images,
        });

        if (!_image.cancelled) {
            setImage(_image.uri);
            updateImages(_image.uri, number)
        }
    };

    const removeImage = async () => {
        setImage('')
        updateImages('', number)
    };

    return (

        <View style={imageUploaderStyles.container}>
            {image ? <TouchableOpacity onPress={removeImage} style={{backgroundColor: '#0f4d92'}}>
                <MaterialIcons name="cancel" size={15} color="white" style={{paddingLeft: '1%', alignSelf: 'flex-end',}} />
            </TouchableOpacity> : null}
            {
                image  ? <Image source={{ uri: image }} style={{ width: '100%', height: '100%' }} />
                : null
            }
        <View style={imageUploaderStyles.uploadBtnContainer}>
            <TouchableOpacity onPress={addImage} style={imageUploaderStyles.uploadBtn} >
                <Text style={{fontSize: 13, color: '#f6f6f6', paddingTop: 3 }}>{image ? 'Edit' : 'Upload'} Image</Text>
                <AntDesign name="camera" size={15} color="#f6f6f6" />
            </TouchableOpacity>
        </View>
        </View>

    );
}

const imageUploaderStyles=StyleSheet.create({
   container:{
       elevation:2,
       height: 140,
       width: '25%',
       backgroundColor:'#f6f6f6',
       position:'relative',
       borderRadius: 5,
       borderColor: 'gray',
       borderWidth: 1,
       overflow:'hidden',
       marginBottom: 10
   },
   uploadBtnContainer:{
       opacity:1,
       position:'absolute',
       right:0,
       bottom:0,
       backgroundColor:'#0f4d92',
       width:'100%',
       height:'28%',
   },
   uploadBtn:{
       display:'flex',
       alignItems:"center",
       justifyContent:'center'
   },
   row: {
    flexDirection: "row",
    flexWrap: "wrap",
  },
})
