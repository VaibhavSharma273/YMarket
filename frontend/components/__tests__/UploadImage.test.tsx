import * as React from 'react';
import renderer from 'react-test-renderer';

import UploadImage from '../UploadImage';

it(`renders correctly`, () => {
    const props = {
        updateImages: jest.fn(),
        defaultValue: jest.fn(),
        number: jest.fn(),
    };
    const tree = renderer.create(<UploadImage {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
});
