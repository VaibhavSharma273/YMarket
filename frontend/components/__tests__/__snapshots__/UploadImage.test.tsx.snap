// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f6f6f6",
      "borderColor": "gray",
      "borderRadius": 5,
      "borderWidth": 1,
      "elevation": 2,
      "height": 140,
      "marginBottom": 10,
      "overflow": "hidden",
      "position": "relative",
      "width": "25%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#0f4d92",
        "bottom": 0,
        "height": "28%",
        "opacity": 1,
        "position": "absolute",
        "right": 0,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#f6f6f6",
            "fontSize": 24,
            "paddingTop": 3,
          }
        }
      >
        Upload
         Image
      </Text>
      <Text />
    </View>
  </View>
</View>
`;
